MariaDB/server
# MDEV-33447: libpmem is not available in RHEL 8

- [x] *The Jira issue number for this PR is: MDEV-33447*
## Description
Because the core Red Hat Enterprise Linux 8 does not include `libpmem`, let us hard-code the necessary part in our code.

`pmem_persist()`: Implement for 64-bit x86, ARM, POWER, RISC-V and Loongarch in a way that should be compatible with the https://github.com/pmem/pmdk/ implementation of `pmem_persist()`.

Note: libpmem had not been previously enabled for Loongarch in our Debian packaging.
## Release Notes
The dependency on `libpmem` was removed by implementing `log_t::pmem_persist()` directly.

On 64-bit systems, the CMake parameter `WITH_INNODB_PMEM` can be set to enable or disable the memory-mapped InnoDB log. It is enabled by default on 64-bit x86, ARM and POWER, and disabled on RISC-V or Loongarch, which are not covered by our current CI infrastructure.
## How can this PR be tested?
I reviewed the generated machine code for the inline assembler snippets related to `pmem_persist()` on https://godbolt.org. It should be noted that GCC blindly copies the output to an external assembler without validating the mnemonics, while `clang` includes a built-in assembler. The GCC output can be validated by enabling a separate option to compile to binary code. This is how the instruction encodings for POWER were validated.

I tested this on two Intel implementations of the AMD64 ISA: one that implements `clflushopt` and `clwb` and another that only implements `clflush`.

This should be covered on all 64-bit CI targets that run tests on Linux `/dev/shm`. Unfortunately, there is no RISC-V or Loongarch coverage in our CI yet, but we do set `WITH_INNODB_PMEM=OFF` by default on those systems.
## Basing the PR against the correct MariaDB version

## PR quality check
